modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/KotlinClientCodegen.java

```diff
@@ -264,6 +291,26 @@ public void processOpts() {
             LOGGER.warn(CodegenConstants.INVOKER_PACKAGE + " with " + this.getName() + " generator is ignored. Use " + CodegenConstants.PACKAGE_NAME + ".");
         }
 
+        if (additionalProperties.containsKey(DATE_LIBRARY)) {
+            setDateLibrary(additionalProperties.get(DATE_LIBRARY).toString());
+        }
+
+        if (DateLibrary.THREETENBP.value.equals(dateLibrary)) {
+            additionalProperties.put(DateLibrary.THREETENBP.value, "true");
+            typeMapping.put("date", "LocalDate");
+            typeMapping.put("DateTime", "LocalDateTime");
+            importMapping.put("LocalDate", "org.threeten.bp.LocalDate");
+            importMapping.put("LocalDateTime", "org.threeten.bp.LocalDateTime");
+        } else if (DateLibrary.STRING.value.equals(dateLibrary)) {
+            additionalProperties.put(DateLibrary.STRING.value, "true");
```

Comment:
I think having just `string` as an additional property would be confusing.

For instance, in a template this would lead to:

```
{{#string}}string date logic{{/string}}{{^string}}other logic{{/string}}
```

This seems like an unlikely use case, until it starts getting used and we can't change it without a breaking change. Also, I think this may cause confusion on some property/params which may have `string` properties as well.

additionalProperties should be true here rather than "true" if it's intended to be used for switching in templates.

Any objection to using `isStringDate` or something? iirc, Mustache will consider the string "true" for properties prefixed with `is` as a boolean. I could be wrong, but I prefer to set booleans explicitly. At one point, we were able to set them as strings but I don't know if that code was changed or if the Mustache dependency was updated but as of a few weeks ago they had to be explicit booleans.